README.md

```diff
@@ -21,44 +21,150 @@
 
 ----
 
-NebulaStream is our attempt to develop a general-purpose, end-to-end data management system for the IoT.
-It provides an out-of-the-box experience with rich data processing functionalities and a high ease-of-use.
+NebulaStream is a post-cloud, end-to-end data-management system for cloud-edge-sensor deployments across the IoT.
+The platform combines Ease of Use, Extensibility, and Efficiency to let teams focus on business logic while the engine optimizes the data path.
 
-NebulaStream is a joint research project between the DIMA group at TU Berlin and BIFOLD.
+**Core Pillars**
+- Ease of Use - Rich, out-of-the-box functionality a variety of stream processing queries.
+- Extensibility - Uniform plugin registries for custom connectors, formats, operators, and optimizations.
+- Efficiency - Hardware-tailored code, adaptive execution, and interleaved I/O to handle large workloads.
 
-Learn more about Nebula Stream at https://www.nebula.stream or take a look at our [documentation](docs).
+## Vision & Architecture
 
-# Clang-Format, Clang-Tidy, License & Pragma Once Check, and Ensure Correct Comments
-We use `clang-format` and `clang-tidy` to ensure code quality and consistency.
-To run the checks, you can use the target `format`. 
+NebulaStream is a joint research project at BIFOLD with contributors from the DIMA Group at TU Berlin and the DFKI IAM Group.
+It aims to execute thousands of queries over millions of heterogeneous sources in massively distributed environments.
+We advance this vision through five core technologies:
+- Heterogeneous Hardware Support - Targets diverse CPU architectures and accelerators such as GPUs and TPUs.
+- Code Generation - Compiles every query to efficient native code for low latency and energy usage.
+- In-Network Processing - Pushes operators toward data sources to reduce network traffic.
+- On-Demand Gathering - Applies processing as early as possible along the data path.
+- Adaptive Resource Management - Reacts to topology or workload changes without interrupting queries.
 
+The system architecture spans sensor to cloud:
+1. Sources & Sinks - Built-in connectors (e.g., JDBC, MQTT, TCP) and formats (e.g., CSV, JSON) with plugin hooks for custom components; extend them via the plugin framework in the [Extensibility guide](docs/guide/extensibility.md).
+2. I/O Handling - Thread-shared source processing and asynchronous callbacks minimize waiting time.
```

Comment:
We are currently still running the thread-per-source model, and do not use callbacks. 
I am not sure what @alepping thinks about whether it is worth starting another effort and integrating the async I/O branch into main after the benchmarks.

Reply:
Yes, I think so!